### ISSUE TITLE
List reverse block

### ISSUE THREAD

**Issue by ewpatton** [mit-cml/appinventor-sources#1347]
We should add a new block to the language for reversing a list. While it is possible to build a procedure to do it using the built-in blocks, it would be better for beginners if the language simply provided it as a built-in operation.

**Comment by kuan9611** [comment]
@ewpatton @josmas Thinking of working on this one. It seems the main additions would be made [here](https://github.com/mit-cml/appinventor-sources/blob/master/appinventor/blocklyeditor/src/blocks/lists.js) and [here](https://github.com/mit-cml/appinventor-sources/blob/master/appinventor/blocklyeditor/src/generators/yail/lists.js). Are there other files I should pay close attention to?

**Comment by ewpatton** [comment]
@kuan9611 A few others pieces to flesh this out:

* blocklyeditor/src/generators/yail.js: Add a constant for the new operation used in the generator
* blocklyeditor/src/msg/en/_messages.js: Add a constant for the translated block name here so it can be internationalized
* buildserver/src/com/google/appinventor/buildserver/resources/runtime.scm: Implement the opeartion here in Scheme (there is a section where all of the list operations are defined)

For the last part (implementing the functionality), note that lists in App Inventor are of the type YailList, which is a linked list where the first element is always the symbol `*list*`, so the body of the reverse operation will look something like:

```scheme
(insert-yail-list-header (reverse (yail-list-contents yail-list)))
```

In addition to the above, you should implement some basic type checking to ensure that yail-list is actually a YailList (see `yail-list?`).

**Comment by kuan9611** [comment]
@ewpatton maybe I'm missing something, but it seems none of the other list operations have a constant defined in [generators/yail.js](https://github.com/mit-cml/appinventor-sources/blob/master/appinventor/blocklyeditor/src/generators/yail.js)?

**Comment by ewpatton** [comment]
@kuan9611 In that case, you can provide the string directly to match the existing code.

**Comment by ewpatton** [comment]
Resolved by b3ce6b5878e592b5e0fd93379eb6e9445ad227c9
